src/graphql/resolveProcedure.js

```diff
@@ -34,13 +39,26 @@ const resolveProcedure = (procedure, getProcedureArgs) => {
   return async (source, args, { client }) => {
     const procedureArgs = getProcedureArgs(source, args)
 
-    // Actuall run the procedure using our arguments.
-    const result = await client.queryAsync({
-      name: query.name,
-      text: query.text,
-      values: argEntries.map(([name]) => procedureArgs[camelCase(name)]),
+    const values = argEntries.map(([name, type]) => {
+      const obj = procedureArgs[camelCase(name)]
+      let value
+
+      if (type.isTableType) {
+        value = reduce(obj, (acc, val, key) => {
```

Comment:
Try instead of using `reduce` and converting keys to an array here, try just using `JSON.stringify`. I’m not sure if it will work, just curious 😊

Another approach may be to convert the object into a PostgreSQL tuple directly. So for example a table that looks like:

```sql
create table test (a int, b int);
```

And an object that looks like:

```json
{
  "a": 1,
  "b": 2
}
```

Would become a tuple that looks like:

```sql
(1,2)
```

The main difference being we remove the JSON middle man and convert directly to a PostgreSQL type. This might be more future proof?

This could also be a bug in [`pg-types`](https://www.npmjs.com/package/pg-types). Thus why I’m curious to see if `JSON.stringify` works.